src/auth_end.ejs

```diff
@@ -7,12 +7,13 @@
     <script>
       const params = new URLSearchParams(window.location.search);
       const result = params.get('code');
-      const { authId, method } = JSON.parse(params.get('state'));
+      const { authId, method, redirectUrl } = JSON.parse(params.get('state'));
```

Comment:
I don't see you decoding the state here. Does URLSearchParams automatically handle that for you?

Reply:
yes, URLSearchParams parses the search params. Encoding is merely for constructing the raw string URL.